internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestHybrid.java

```diff
@@ -271,6 +274,165 @@ public void testHybridInitializationOnMultiColo() throws IOException {
+              TestUtils.getVeniceWriterFactory(veniceWriterProperties2, pubSubProducerAdapterFactory)
+                  .createVeniceWriter(new VeniceWriterOptions.Builder(versionTopicName).build())) {
+        veniceWriter1.broadcastStartOfPush(false, Collections.emptyMap());
+
+        /**
+         * Explicitly simulate split-brain issue.
+         * Writer1:
+         *
+         * key_0: value_0 with upstream offset: 5
+         * key_1: value_1 with upstream offset: 6
+         * key_2: value_2 with upstream offset: 7
+         * key_3: value_3 with upstream offset: 8
+         * key_4: value_4 with upstream offset: 9
+         * Writer2:
+         * key_0: value_x with upstream offset: 3
+         * key_5: value_5 with upstream offset: 10
+         * key_6: value_6 with upstream offset: 11
+         * key_7: value_7 with upstream offset: 12
+         * key_8: value_8 with upstream offset: 13
+         * key_9: value_9 with upstream offset: 14
+         */
+
+        // Sending out dummy records first to push out SOS messages first.
+        veniceWriter1.put(
+            stringSerializer.serialize("key_writer_1"),
+            stringSerializer.serialize("value_writer_1"),
+            1,
+            null,
+            new LeaderMetadataWrapper(0, 0));
+        veniceWriter1.flush();
+        veniceWriter2.put(
+            stringSerializer.serialize("key_writer_2"),
+            stringSerializer.serialize("value_writer_2"),
+            1,
+            null,
+            new LeaderMetadataWrapper(1, 0));
+        veniceWriter2.flush();
+
+        for (int i = 0; i < 5; ++i) {
+          veniceWriter1.put(
+              stringSerializer.serialize("key_" + i),
+              stringSerializer.serialize("value_" + i),
+              1,
+              null,
+              new LeaderMetadataWrapper(i + 5, 0));
+        }
+        veniceWriter1.flush();
+        veniceWriter2.put(
+            stringSerializer.serialize("key_" + 0),
+            stringSerializer.serialize("value_x"),
+            1,
+            null,
+            new LeaderMetadataWrapper(3, 0));
+        for (int i = 5; i < 10; ++i) {
+          veniceWriter2.put(
+              stringSerializer.serialize("key_" + i),
+              stringSerializer.serialize("value_" + i),
+              1,
+              null,
+              new LeaderMetadataWrapper(i + 5, 0));
+        }
+        veniceWriter2.flush();
+        veniceWriter1.broadcastEndOfPush(Collections.emptyMap());
+        veniceWriter1.flush();
+      }
+
+      TestUtils.waitForNonDeterministicAssertion(100, TimeUnit.SECONDS, true, () -> {
+        // Now the store should have version 1
+        JobStatusQueryResponse jobStatus = controllerClient.queryJobStatus(Version.composeKafkaTopic(storeName, 1));
+        Assert.assertFalse(jobStatus.isError(), "Error in getting JobStatusResponse: " + jobStatus.getError());
+        assertEquals(jobStatus.getStatus(), "ERROR");
```

Comment:
I'm a bit confused with the test assertion. Does the push job succeed with the split brain problem? If so, why job status is ERROR? If not, why `jobStatus.isError()` is false?

Reply:
The push job should fail because of the split-brain issue before reporting completed.
`jobStatus.isError` is to check whether Controller is serving the metadata request or not.
Even the job status is `ERROR`, `JobStatusQueryResponse` itself shouldn't be error.

Reply:
Gotcha, thanks!

Reply:
Gotcha, thanks!